agent/mapcall/src/main/java/randoop/MethodReplacements.java

```diff
@@ -0,0 +1,42 @@
+package randoop;
+
+import java.lang.instrument.Instrumentation;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class used to communicate the signatures of methods replaced by the mapcalls agent to Randoop so
+ * Randoop can omit direct calls to these methods.
+ *
+ * <p>Uses synchronized access to a static list of the signature strings, because the mapcalls agent
+ * can run on multiple classes concurrently.
+ *
+ * <p>This list is set by {@link randoop.instrument.MapCallsAgent#premain(String, Instrumentation)}
+ * before the {@link randoop.instrument.CallReplacementTransformer} is added to the class loader,
+ * and the method {@link #addReplacedMethods(List)} should only be called at that point. Randoop
+ * should add the result of {@link #getSignatureList()} to the {@code --omitmethods} patterns before
+ * starting generation, because any method that is being replaced shouldn't be directly called.
+ */
+public class MethodReplacements {
```

Comment:
Some of these details have to do with the configuration of the jar files. I'm trying to make it so that the agent doesn't depend on Randoop.

We can try tweaks before doing the release, but I'm leaving this alone for now.